hack/tools/go.mod

```diff
@@ -1,50 +1,60 @@
 module sigs.k8s.io/kind/hack/tools
 
-go 1.23
+go 1.23.0
+
+toolchain go1.24.0
```

Comment:
@BenTheElder where are we now on the use of `toolchain` ? I remember we discussed this in the past but can't remember what we have decided

Reply:
Does that apply for `tools/` too though? I thought we treated that a little differently than the root `go.mod`.

Mostly just curious. Updating to drop this part of the change as I don't think it's really related or necessary for the main goal.

Reply:
> Does that apply for tools/ too though? I thought we treated that a little differently than the root go.mod.

These tools are primarily applied to go code that we'll build at .go-version, I'm not sure if any of them will switch behavior if they assume N+1 go version.

Reply:
It's easier to reason about matching them, if possible.

Reply:
apparently there is a go bug that will be fixed about inserting toolchain when it wasn't specified before

Reply:
Oops, I missed excluding this in the latest update. I'm not using 1.25 yet, and I agree we shouldn't move to it quite yet.

Will remove this line and push an update.